source/DataWriter.cpp

```diff
@@ -55,22 +49,51 @@ DataWriter::~DataWriter()
 
 
 
+// Gets the current contents of the DataWriter.
+string DataWriter::GetText()
+{
+	return out.str();
+}
+
+
+
 // Save the contents to a file.
-void DataWriter::SaveToPath(const std::string &filepath)
+void DataWriter::SaveToPath(const string &filepath)
 {
 	Files::Write(filepath, out.str());
 }
 
 
 
+// Writes the contents of the string without any escaping, quoting or
+// any other kind of modification.
+DataWriter &DataWriter::WriteRaw(const string &c)
```

Comment:
This function is not really necessary, since it doesn't need to be handled separately than the template.

Reply:
Should I just remove the static assert from the template?

Reply:
yes